pkg/render/logstorage.go

```diff
@@ -522,22 +526,32 @@ func (es elasticsearchComponent) elasticsearchCluster() *esv1.Elasticsearch {
 			NodeSets: []esv1.NodeSet{
 				{
 					Count: int32(nodeConfig.Count),
-					Name:  "es",
+					Name: nodeSetName(pvcTemplate),
 					Config: &cmnv1.Config{
 						Data: map[string]interface{}{
 							"node.master": "true",
 							"node.data":   "true",
 							"node.ingest": "true",
 						},
 					},
-					VolumeClaimTemplates: []corev1.PersistentVolumeClaim{es.pvcTemplate()},
+					VolumeClaimTemplates: []corev1.PersistentVolumeClaim{pvcTemplate},
 					PodTemplate:          es.podTemplate(),
 				},
 			},
 		},
 	}
 }
 
+// nodeSetName returns thumbprint of PersistentVolumeClaim object as string.
+// As storage requirements of NodeSets are immutable,
+// renaming a NodeSet automatically creates a new StatefulSet with new PersistentVolumeClaim.
+// https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-orchestration.html#k8s-orchestration-limitations
+func nodeSetName(pvcTemplate corev1.PersistentVolumeClaim) string{
+	pvcTemplateHash := fnv.New64a()
+	templateBytes, _ := json.Marshal(pvcTemplate)
```

Comment:
This returns an error and it should be handled. I think the pattern in render is to panic, since we don't return errors. If there's an error the below line likely panicsince templateBytes will be nil.

Reply:
Maybe confirm with @caseydavenport or @tmjd that this is OK, sometimes our opinions with panicing differ. This needs to be handle somehow though.

Reply:
blech, yeah it's probably fine.

I don't really like the pattern that we've established to "panic" in render - it's not very graceful or resilient to potential bugs in the code.

Ideally we'd propagate an error back up to report to the user in the TigeraStatus, but that's a larger change than I want to make you do in this PR.

Something better but not as good would be to log out a warning and then return sensible output. e.g., if we can't calculate a hash, we just return `es` like we do today?

Reply:
>Something better but not as good would be to log out a warning and then return sensible output. e.g., if we can't calculate a hash, we just return es like we do today?

Sure sounds better than panicking.